engine/src/main/java/com/datatorrent/stram/engine/GenericNode.java

```diff
@@ -608,9 +610,9 @@ else if (tracker.ports[trackerIndex] == null) {
           if (need2sleep) {
             if (handleIdleTime && insideWindow) {
               ((IdleTimeHandler) operator).handleIdleTime();
-            }
-            else {
+            } else {
               Thread.sleep(spinMillis);
+              spinMillis = Math.min(maxSpinMillis, spinMillis + 1);
```

Comment:
I had looked at this before in a difference scenario. Can we do a better heuristic. We can have another timeout setting (which is less than streaming window size) where we continue to do busy way till that timeout and if there is no data till then, then initiate the stepwise sleep.